src/v/cluster/types.cc

```diff
@@ -654,7 +655,7 @@ adl<model::timeout_clock::duration>::from(iobuf_parser& in) {
     // because rep has type of long long
     // auto rp = adl<rep>{}.from(in);
     auto rp = adl<uint64_t>{}.from(in);
-    return duration(rp);
+    return std::chrono::duration_cast<duration>(std::chrono::milliseconds{rp});
```

Comment:
I've written some tests. Which overload gets picked up depends on whether `cluster/types.h` is included.

They seem to handle signed and unsigned perfectly well due to 2's complement overflow.

The old version isn't broken since it was milliseconds. The new version makes that explicit.

In general, I think this is pretty scary code.